finality-aleph/src/network.rs

```diff
@@ -470,27 +573,20 @@ impl<D: Clone + Encode + Decode, B: BlockT + 'static, N: Network<B> + Clone>
             Meta(message, recipient) => {
                 let message = InternalMessage::Meta(message);
                 match recipient {
-                    None => {
+                    Recipient::All => {
                         for (peer_id, _) in self.peers.lock().all_peers.iter() {
                             self.send_message(peer_id, message.clone());
                         }
                     }
-                    Some(peer_id) => self.send_message(&peer_id, message),
+                    Recipient::Target(peer_id) => self.send_message(&peer_id, message),
                 }
             }
             Data(session_id, data, recipient) => {
-                let message = InternalMessage::Data(session_id, data);
-                match recipient {
-                    None => {
-                        for peer_id in self.peers.lock().peers_authenticated_for(session_id) {
-                            self.send_message(peer_id, message.clone());
-                        }
-                    }
-                    Some(node_id) => {
-                        if let Some(peer_id) = self.peers.lock().get(session_id, node_id) {
-                            self.send_message(peer_id, message);
-                        }
-                    }
+                let msg_bytes = data.encode();
+                let index = self.check_or_insert_cache(session_id, msg_bytes);
+                let message = InternalMessage::Data(session_id, index, data, recipient);
+                for peer_id in self.peers.lock().peers_authenticated_for(session_id) {
+                    self.send_message(peer_id, message.clone());
```

Comment:
Am I reading this correctly that you always send the message to everyone, even if it is addressed to one of the nodes you actually know?

Reply:
Fixed.